src/components/application_manager/src/message_helper.cc

```diff
@@ -2395,24 +2395,21 @@ mobile_apis::Result::eType MessageHelper::ProcessSoftButtons(
     }
 
     switch (request_soft_buttons[i][strings::type].asInt()) {
-      case SoftButtonType::SBT_IMAGE: {
-        // Any text value for type "IMAGE" should be ignored.
-        if (request_soft_buttons[i].keyExists(strings::text)) {
-          request_soft_buttons[i].erase(strings::text);
-        }
-
-        if ((!request_soft_buttons[i].keyExists(strings::image) ||
-             (Result::SUCCESS !=
-              VerifyImage(
-                  request_soft_buttons[i][strings::image], app, app_mngr)))) {
-          return Result::INVALID_DATA;
-        }
-        break;
-      }
+      case SoftButtonType::SBT_IMAGE:
       case SoftButtonType::SBT_TEXT: {
-        if ((!request_soft_buttons[i].keyExists(strings::text)) ||
-            (!VerifySoftButtonString(
-                 request_soft_buttons[i][strings::text].asString()))) {
+        Result::eType verification_result = Result::INVALID_DATA;
+        verification_result =
+            VerifyImage(request_soft_buttons[i][strings::image], app, app_mngr);
+        const bool is_image_verified =
+            (request_soft_buttons[i].keyExists(strings::image)) &&
```

Comment:
@indlu, why do you check whether key exists after you already tried to access it?
In any case, because you tried to get `image` before this statement is always be true. It because `request_soft_buttons` is not a const reference. As result `SmartObject` in case of key absence will add it with NULL type.